packages/Sandblocks-Smalltalk/SBStNewMessageSend.class.st

```diff
@@ -245,8 +245,16 @@ SBStNewMessageSend >> pc: aNumber [
 { #category : #'as yet unclassified' }
 SBStNewMessageSend >> possibleArgumentNamesFor: aString [
 
-	(self systemNavigation allImplementorsOf: aString asSymbol) do: [:impl | ^ impl compiledMethod asSandblock arguments collect: #contents].
-	^ #()
+	| selector method |
+	selector := aString asSymbol.
+	self owner receiver guessClassExpensive ifNotNil: [:class | (class lookupSelector: selector) ifNotNil: [:meth | method := meth]].
```

Comment:
Note that I used `guessClassExpensive` here because in my examples, `guessedClass` was nil. Any chance to improve caching?